lib/replset.js

```diff
@@ -158,6 +182,15 @@ class ReplSet extends EventEmitter {
         // Get the version information
         var result = yield self.discover();
 
+        // Set protocol version for replicaset to be given value in options or
+        // default to version 1 if using MongoDB 3.2 or above
+        // Protocol version 1 is supported since 3.2+ and required for 4.0+
+        var protocolVersion = self.protocolVersion;
+        if (typeof protocolVersion !== 'number') {
+          protocolVersion =
+            result.version[0] >= 4 || (result.version[0] === 3 && result.version[1] >= 2) ? 1 : 0;
```

Comment:
Would it make sense to introduce a dependency on `semver` so that we can make server version checks more readable? We already depend on it in e.g. `mongodb-version-manager` so this wouldn't really affect the whole dependency graph that much

Reply:
:+1: